### PR TITLE
Add support for binning of nested queries

### LINKED ISSUE #5938: Binning isn't available when using nested questions
On Chome on release-0.26.0. Not sure if this was known/expected.

Normal query, which allows binning this field:

screen shot 2017-09-11 at 4 12 08 pm

That same table, but as a saved question used as a nested query — notice the binning options aren't there:

screen shot 2017-09-11 at 4 12 51 pm

### PR DESCRIPTION
**Pull request by senior**
To support this Cards now include fingerprints in the
result_metadata. That fingerprint is included in the query dictionary
and is available to the query pipeline. The expand/resolve/binning
middleware can make use of that fingerprint data to build the binned
query. Dimension options for these kinds of queries will also now be
available.

Note that binning is only available on questions that have result
metadata. If no result metadata (and thus no fingerprint) is
available, no dimensions options will be included and an exception
will be thrown if a binned query is attempted. In most cases, just
running a card once will populate that data.

Fixes #5938

### CODE REVIEW

File: src/metabase/sync/analyze/query_results.clj [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                      "Valid array of results column metadata maps")
+    "value must be an array of valid results column metadata maps."))
+
+(s/defn ^:private maybe-infer-special-type :- ResultColumnMetadata
+  "If we already have a special type from the stored metadata, use that, otherwise compute the special type similar to
+  how our sync process would compute it for a new field."
+  [result-metadata col]
+  (if (contains? result-metadata :special_type)
+    result-metadata
+    (assoc result-metadata :special_type (classify-name/infer-special-type col))))
```
Comment (sbelak) [inline src/metabase/sync/analyze/query_results.clj]: infer-special-type already has rules when to rerun. I think it's better to defer to it completely.

**Review by sbelak — APPROVED** [review]
LGTM

File: src/metabase/sync/analyze/query_results.clj (hunk 1 above)
Comment (senior) [reply src/metabase/sync/analyze/query_results.clj]: Do you mean there's no need to run it here? What about in the case of native queries where it wouldn't have ran at all?

Comment (sbelak) [reply src/metabase/sync/analyze/query_results.clj]: Sorry I was unclear. I meant just the check. The whole fn can collapse into just 
```
(assoc result-metadata :special_type (classify-name/infer-special-type col))
```

**Pull request merged**